### DISCUSSION TITLE
secret-tool attributes

### DISCUSSION THREAD

**Discussion by grawity** [pdobsan/oama#25] [Ideas]
In recent commits, it seems that the `mailctl` attribute is used both as a search key _and_ to distinguish mailctl entries from other entries. It would probably be better to use the schema ID for the latter purpose – it can be stored as the `xdg:schema` attribute (although `secret-tool` and libsecret handle it differently, in the underlying D-Bus API it's really just an attribute).

```
secret-tool store --label="foo" \
    xdg:schema "io.github.pdobsan.Mailctl.Account" \
    account "foo@example.com" ;

secret-tool lookup \
    xdg:schema "io.github.pdobsan.Mailctl.Account" \
    ...
```

**Comment by pdobsan** [comment]
Thank you for raising this idea. I must admit I had to perform considerable
research just to understand what you were talking about then more to judge
whether your suggestion would make anything better.

I don't like at all what I have learned. The whole schema thing is just a
convoluted convention, sparsely documented if documented at all,
inconsistent, confusing. Even the gnome programs secret-tool, seahorse,
key-ring etc. libsecret use it in different ways frequently _hidden_ behind
the user's back. For example xdg:schema is always added regardless of the
user asked for it or did not. The value of this hidden attribute varies
depending on how the item was created!

So my conclusion (at least for the time being) is that the best to keep
clear from these schema/gnome things and just use my own self explanatory
label, attribute name/values.
